nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/reporting/ReportLineageToAtlas.java

```diff
@@ -385,31 +401,50 @@ protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(String proper
+        final SSLContextService sslContextService = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
         final ValidationResult.Builder invalidSSLService = new ValidationResult.Builder()
-                .subject(KAFKA_SSL_CONTEXT_SERVICE.getDisplayName()).valid(false);
+                .subject(SSL_CONTEXT_SERVICE.getDisplayName()).valid(false);
 
+        AtomicBoolean isAtlasApiSecure = new AtomicBoolean(false);
         String atlasUrls = context.getProperty(ATLAS_URLS).evaluateAttributeExpressions().getValue();
         if (!StringUtils.isEmpty(atlasUrls)) {
             Arrays.stream(atlasUrls.split(ATLAS_URL_DELIMITER))
                 .map(String::trim)
                 .forEach(input -> {
-                    final ValidationResult.Builder builder = new ValidationResult.Builder().subject(ATLAS_URLS.getDisplayName()).input(input);
                     try {
-                        new URL(input);
-                        results.add(builder.explanation("Valid URI").valid(true).build());
+                        final URL url = new URL(input);
+                        if ("https".equalsIgnoreCase(url.getProtocol())) {
+                            isAtlasApiSecure.set(true);
+                        }
                     } catch (Exception e) {
-                        results.add(builder.explanation("Contains invalid URI: " + e).valid(false).build());
+                        results.add(new ValidationResult.Builder().subject(ATLAS_URLS.getDisplayName()).input(input)
+                                .explanation("contains invalid URI: " + e).valid(false).build());
                     }
                 });
         }
 
+        if (isAtlasApiSecure.get()) {
+            if (sslContextService == null) {
+                results.add(invalidSSLService.explanation("required for connecting to Atlas via HTTPS.").build());
+            } else if (context.getControllerServiceLookup().isControllerServiceEnabled(sslContextService)) {
+                if (!sslContextService.isTrustStoreConfigured()) {
+                    results.add(invalidSSLService.explanation("no truststore configured which is required for connecting to Atlas via HTTPS.").build());
+                } else if (!KEYSTORE_TYPE_JKS.equalsIgnoreCase(sslContextService.getTrustStoreType())) {
+                    results.add(invalidSSLService.explanation("truststore type is not JKS. Atlas client supports JKS truststores only.").build());
+                }
+            }
+        }
+
         final String atlasAuthNMethod = context.getProperty(ATLAS_AUTHN_METHOD).getValue();
         final AtlasAuthN atlasAuthN = getAtlasAuthN(atlasAuthNMethod);
         results.addAll(atlasAuthN.validate(context));
 
-
-        namespaceResolverLoader.forEach(resolver -> results.addAll(resolver.validate(context)));
+        synchronized (namespaceResolverLoader) {
```

Comment:
`customValidate()` runs before `@OnScheduled` gets called so that point is to late.
I tried `init()` but ran into classloader issues with NamespaceResolver interface and its implementations.